modules/extensions-support/src/main/java/org/mule/module/extension/internal/introspection/describer/AnnotationsBasedDescriber.java

```diff
@@ -308,16 +323,26 @@ private FieldDescriber getFieldDescriber(Field field)
 
     private void declareOperations(DeclarationDescriptor declaration, Class<?> extensionType)
     {
-        Operations operations = extensionType.getAnnotation(Operations.class);
-        if (operations != null)
+        Class<?>[] operations = getOperationClasses(extensionType);
+        for (Class<?> actingClass : operations)
         {
-            for (Class<?> actingClass : operations.value())
+            if (actingClass.isAssignableFrom(extensionType))
             {
-                declareOperation(declaration, actingClass);
+                throw new IllegalModelDefinitionException(String.format(
+                        "Operation class '%s' can't be assignable from extension class '%s'",
```

Comment:
this error message is not meaningful. What it should explain ot the user is that a class which serves as a config cannot contain operations